components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/event/queue/EventListMapManager.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.stream.processor.core.event.queue;
+
+import org.apache.log4j.Logger;
+import org.wso2.carbon.stream.processor.core.ha.exception.InvalidByteMessageException;
+import org.wso2.carbon.stream.processor.core.ha.tcp.SiddhiEventConverter;
+import org.wso2.carbon.stream.processor.core.ha.util.HAConstants;
+import org.wso2.carbon.stream.processor.core.internal.SiddhiAppData;
+import org.wso2.carbon.stream.processor.core.internal.StreamProcessorDataHolder;
+import org.wso2.carbon.stream.processor.core.util.BinaryMessageConverterUtil;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.source.Source;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+public class EventListMapManager {
+    private static ConcurrentSkipListMap<Long,QueuedEvent> eventListMap;
+    private static Map<String,Long> perAppLastControlMessageSequenceNumberList = new HashMap<>();
+    private static final Logger log = Logger.getLogger(EventListMapManager.class);
+    private static long startTime;
+    private static long endTime;
+    private static int count = 0;
+    private static final int TPS_EVENT_THRESHOLD = 100000;
+
+    public EventListMapManager() {
+    }
+
+    public static void initializeEventListMap() {
+        eventListMap = new ConcurrentSkipListMap<Long, QueuedEvent>();
+    }
+
+    public void parseControlMessage(byte[] controlMessageContentByteArray) throws UnsupportedEncodingException {
+
+        String message = null;
+        try {
+            message = new String(controlMessageContentByteArray, HAConstants.DEFAULT_CHARSET);
+        } catch (UnsupportedEncodingException e) {
+            throw new UnsupportedEncodingException();
```

Comment:
No need of the try catch if we are throwing the same exception

Reply:
ah yes